ui/app/appointments/controllers/manage/calendar/appointmentsCalendarViewController.js

```diff
@@ -1,8 +1,12 @@
 'use strict';
 
 angular.module('bahmni.appointments')
-    .controller('AppointmentsCalendarViewController', ['$scope', '$state', '$translate', 'spinner', 'appointmentsService', 'appointmentsFilter', '$rootScope',
-        function ($scope, $state, $translate, spinner, appointmentsService, appointmentsFilter, $rootScope) {
+    .controller('AppointmentsCalendarViewController', ['$scope', '$state', '$translate', 'spinner', 'appointmentsService', 'appointmentsFilter', '$rootScope', '$interval', 'appService',
+        function ($scope, $state, $translate, spinner, appointmentsService, appointmentsFilter, $rootScope, $interval, appService) {
+            var autoRefreshIntervalInSeconds = parseInt(appService.getAppDescriptor().getConfigValue('autoRefreshIntervalInSeconds'));
```

Comment:
I think you should also add a check to accept only positive numbers or if its negative(an unlikely situation), convert the negative number to positive

Reply:
By default negative numbers will be handled in the interval function. If we give any negative number then the setInterval function will  automatically set 4ms time as interval time which is minimum default value

Reply:
> By default negative numbers will be handled in the interval function. If we give any negative number then the setInterval function will automatically set 4ms time as interval time which is minimum default value

Ok  cool. Everything is fine then